sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/BuilderAnnotationVisitor.java

```diff
@@ -72,7 +72,7 @@ public void visitClass(ClassElement element, VisitorContext context) {
             TypeDef propertyTypeDef = TypeDef.of(beanProperty.getType());
             TypeDef fieldType = propertyTypeDef.makeNullable();
             if (!fieldType.isNullable()) {
-                throw  new IllegalStateException();
+                throw  new IllegalStateException("Could not make the field nullable");
```

Comment:
Had to create a test to debug this, as `IllegalStateException` with no message was not descriptive enough